File: examples/v2/insights-account/go/main.go
package main

import (
	insights "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/insights"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		accountSuffix := cfg.Get("accountSuffix")
		if accountSuffix == "" {
			accountSuffix = "dev"
		}
		insightsEnv := cfg.Get("insightsEnvironment")
		if insightsEnv == "" {
			insightsEnv = "insights/credentials"
		}

		account, err := insights.NewAccount(ctx, "account", &insights.AccountArgs{
			OrgName:      pulumi.String(serviceOrg),
			AccountName:  pulumi.String("v2-insights-" + accountSuffix),
			Provider:     pulumi.String("aws"),
			Environment:  pulumi.String(insightsEnv),
			ScanSchedule: pulumi.String("none"),
		})
		if err != nil {
			return err
		}

		if _, err := insights.NewScheduledScanSettings(ctx, "scanSettings", &insights.ScheduledScanSettingsArgs{
			OrgName:      pulumi.String(serviceOrg),
			AccountName:  account.AccountName,
			Paused:       pulumi.Bool(true),
			ScheduleCron: pulumi.String("0 6 * * *"),
		}); err != nil {
			return err
		}

		ctx.Export("accountName", account.AccountName)
		return nil
	})
}

File: examples/v2/insights-account/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
organization_name = config.get("organizationName") or "service-provider-test-org"
account_suffix = config.get("accountSuffix") or "dev"
insights_environment = config.get("insightsEnvironment") or "insights/credentials"

account_name_value = f"v2-insights-{account_suffix}"
account = ps_v2.insights.Account(
    "account",
    org_name=organization_name,
    account_name=account_name_value,
    provider="aws",
    environment=insights_environment,
    scan_schedule="none",
)

ps_v2.insights.ScheduledScanSettings(
    "scanSettings",
    org_name=organization_name,
    account_name=account_name_value,
    paused=True,
    schedule_cron="0 6 * * *",
    opts=pulumi.ResourceOptions(depends_on=[account]),
)

pulumi.export("accountName", account.name)
